src/components/scenes/WalletListScene.js

```diff
@@ -73,29 +72,31 @@ type DispatchProps = {
 type Props = StateProps & DispatchProps
 
 type State = {
-  sorting: boolean,
-  showOtpResetModal: boolean,
-  showMessageModal: boolean,
-  messageModalMessage: ?string
+  sorting: boolean
 }
 
 class WalletListComponent extends Component<Props, State> {
   constructor (props: Props) {
     super(props)
     slowlog(this, /.*/, global.slowlogOptions)
     this.state = {
-      sorting: false,
-      showOtpResetModal: this.props.otpResetPending,
-      showMessageModal: false,
-      messageModalMessage: null
+      sorting: false
     }
   }
 
-  UNSAFE_componentWillReceiveProps (nextProps: Props) {
-    if (nextProps.otpResetPending && nextProps.otpResetPending !== this.props.otpResetPending) {
-      this.setState({
-        showOtpResetModal: true
-      })
+  async componentDidMount () {
+    if (this.props.otpResetPending) {
+      const resolved = await Airship.show(bridge => (
+        <TwoButtonSimpleConfirmationModal
+          bridge={bridge}
+          icon={<Image source={otpIcon} />}
+          title={s.strings.otp_modal_reset_headline}
+          subTitle={s.strings.otp_modal_reset_description}
+          cancelText={s.strings.request_review_answer_no}
+          doneText={s.strings.request_review_answer_yes}
+        />
+      ))
+      resolved ? this.props.keepOtp() : this.props.disableOtp()
```

Comment:
We could separate the modal creation into a separate function? So if ever we have other logic on componentDidMount it won't be cluttered.

Reply:
I like that idea. Let's do it.